test/integration/targets/vmware_guest_register_operation/tasks/main.yml

```diff
@@ -0,0 +1,146 @@
+# Test code for the vmware_guest_register_operation module
+# Copyright: (c) 2019, sky-joker <sky.jokerxx@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- import_role:
+    name: prepare_vmware_tests
+  vars:
+    setup_attach_host: true
+    setup_datastore: true
+    setup_cluster: true
+    setup_virtualmachines: true
+
+- name: get a list of Resource pools from vcsim
+  uri:
+    url: http://{{ vcenter_hostname }}:5000/govc_find?filter=RP
+  register: resource_pools
+
+- name: get a resource pool
+  set_fact: resource_pool="{{ resource_pools['json'][2].split('/').7 }}"
+
+- name: get a vm name
+  set_fact: vm_name="DC0_H0_VM0"
```

Comment:
Two options here:

- You use `setup_virtualmachines: true` to let `prepare_vmware_tests` prepare the VM for you, and you can use "{{ virtual_machines[0].name }}" or "{{ virtual_machines_in_cluster[0].name }}"
- OR, you remove `setup_virtualmachines: true` the you prepare the VM yourself (please just call it test_vm1 https://docs.ansible.com/ansible/devel/dev_guide/platforms/vmware_guidelines.html#vm-names-should-be-predictable)